### PR TITLE
20 login UI

### LINKED ISSUE #20: Create a login activity with UI to sign in with Google and with an email and password

### PR DESCRIPTION
**Pull request by clementcharmillot**
Adds a new activity and two fragments to login and register.

### Login UI:

closes #20

### CODE REVIEW

File: app/src/main/java/com/github/drawyourpath/bootcamp/login/LoginActions.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val view = super.onCreateView(inflater, container, savedInstanceState);
+
+        val registerButton = view?.findViewById<Button>(R.id.BT_Register);
+        registerButton?.setOnClickListener { viewModel.showRegisterUI() }
+
+        return view
+    }
```
Comment (HoLuDyEr) [inline app/src/main/java/com/github/drawyourpath/bootcamp/login/LoginActions.kt:31]: from the documentation of onCreateView, this method should only be about changing the inflater, and logic for the view should be in function :
`override fun onViewCreated(view: View, savedInstanceState: Bundle?) {}`
so you should move 
`val registerButton = view?.findViewById<Button>(R.id.BT_Register);`
`registerButton?.setOnClickListener { viewModel.showRegisterUI() }`
inside onViewCreated

File: app/src/main/java/com/github/drawyourpath/bootcamp/login/RegisterActions.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        savedInstanceState: Bundle?
+    ): View? {
+        val view = super.onCreateView(inflater, container, savedInstanceState);
+
+        val registerButton = view?.findViewById<Button>(R.id.BT_Login);
+
+        registerButton?.setOnClickListener { viewModel.showLoginUI() }
+
+        return view
+    }
```
Comment (HoLuDyEr) [inline app/src/main/java/com/github/drawyourpath/bootcamp/login/RegisterActions.kt:31]: same as loginAction
view logic should be moved inside 
`override fun onViewCreated(view: View, savedInstanceState: Bundle?) {}`

File: app/src/main/res/layout/activity_login.xml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            android:id="@+id/SP_Actions"
+            android:layout_width="match_parent"
+            android:layout_height="144dp"
+            android:layout_weight="1" />
+
+        <androidx.fragment.app.FragmentContainerView
+            android:id="@+id/fragment_container_view"
+            android:name="com.github.drawyourpath.bootcamp.login.RegisterActions"
+            android:layout_width="match_parent"
+			android:layout_height="wrap_content"
```
Comment (HoLuDyEr) [inline app/src/main/res/layout/activity_login.xml:53]: little detail: reformat code

File: app/src/main/res/layout/fragment_login_actions.xml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+			android:hint="Email"
+            android:ems="10"
+            android:inputType="textEmailAddress" />
+
+        <EditText
+            android:id="@+id/ET_Password"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:ems="10"
+			android:hint="Password"
```
Comment (HoLuDyEr) [inline app/src/main/res/layout/fragment_login_actions.xml:26]: little detail: reformat code

File: app/src/main/res/layout/fragment_login_actions.xml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:orientation="vertical">
+
+        <EditText
+            android:id="@+id/ET_Email"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+			android:hint="Email"
```
Comment (HoLuDyEr) [inline app/src/main/res/layout/fragment_login_actions.xml:17]: little detail: reformat code

File: app/src/main/res/layout/fragment_register_actions.xml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:orientation="vertical">
+
+        <EditText
+            android:id="@+id/ET_Email"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+			android:hint="Email"
```
Comment (HoLuDyEr) [inline app/src/main/res/layout/fragment_register_actions.xml:17]: little detail: reformat code

File: app/src/main/res/layout/fragment_register_actions.xml [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+            android:ems="10"
+            android:inputType="textEmailAddress" />
+
+        <EditText
+            android:id="@+id/ET_Password"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+			android:hint="Password"
```
Comment (HoLuDyEr) [inline app/src/main/res/layout/fragment_register_actions.xml:25]: little detail: reformat code

**Review by HoLuDyEr — CHANGES_REQUESTED** [review]
overall it is a good PR, it just needs a quick fix

File: app/src/main/java/com/github/drawyourpath/bootcamp/login/LoginActions.kt (hunk 1 above)
Comment (clementcharmillot) [reply app/src/main/java/com/github/drawyourpath/bootcamp/login/LoginActions.kt]: Thanks for the information. I've switched the method used to `onViewCreated()`.

File: app/src/main/res/layout/activity_login.xml (hunk 3 above)
Comment (clementcharmillot) [reply app/src/main/res/layout/activity_login.xml:53]: Nice catch, I've replaced these tabs with spaces.

**Review by HoLuDyEr — APPROVED** [review]
everything is resolved, it can be merged

**Pull request merged**